File: data_structures/array/skeleton/lib/static_array.rb
# This class just dumbs down a regular Array to be staticly sized.
class StaticArray
  def initialize(length)
    self.store = Array.new(length, nil)
    # REVIEW how to make sure When copying an array the pointers aren't putting to the same array
    # deep dup
  end

  # O(1)
  def [](index)
    # get
    return store[index]
  end

  # O(1)
  def []=(index, value)
    # set
    store[index] = value
  end

  protected
  attr_accessor :store
end

File: data_structures/array/static_array.js
class StaticArray {
  constructor(length) {
    this.store = new Array(length).fill(null);
  }

  get(index) {
    return this.store[index]
  }

  set(index, value) {
    return this.store[index] = value;
  }
}

export default StaticArray;
